### DISCUSSION TITLE
RangeSlider wrong value on RangeSelectionChanged event

### DISCUSSION THREAD

**Discussion by guyber** [MahApps/MahApps.Metro#4324] [Q&A] [answered]
hi,

I noticed that the NewUpperValue is not updated correctly in the RangeSelectionChanged event.
The scenario - a RangeSlider set with value 0-9999, when moving the range selection upper value to 14.2% (1/7) of the slider the NewUpperValue  shows the value of ~1600, now drag the range so the lower side is on the same position (14.2%) - the value is around ~1400 - seems like the NewUpperValue is driffted for some reason - maybe related o the range between the upper and lower ends - but looks like a bit more complicated issue.
![range slider](https://user-images.githubusercontent.com/7201649/177160596-fe8e79a5-b9a9-4af9-937f-dff7a52f8cd2.png)

**Comment by punker76** [comment, marked as answer]
The two values disagreed about the same place on the track, and your screenshots say it plainly:
the upper thumb reports 1628 where the lower one, standing on that very spot, reports 1382.

The two thumbs took their own width out of the track and were laid out beside one another, so the
span the values were spread over was shorter than the track the ticks were drawn on, and it started
at a different place for each of them. A thumb hangs over the ends by half its width now and takes
nothing out of the track, so its middle is the point its value stands for. Either thumb dragged onto
a place on the track now lands on the value that place is worth, and `RangeSelectionChanged` hands
over what the thumb stands on.

Held against a slider from 0 to 9999 like yours, a seventh along the track was worth 1209 from the
upper thumb and 1444 from the lower one, and both come to 1428 now.

This is on `develop` and not in a release yet. Thanks for the report and for the picture, it is what
made the two numbers comparable.
